### ISSUE TITLE
nu panicks/crashes if glob stumbles over files w/ unexpected encoding

### ISSUE THREAD

**Issue by joergsch** [nushell/nushell#10640]
### Describe the bug

`glob *` crashes the shell if it stumbles of files w/ unexpected encoding

full backtrace:
```
thread 'main' panicked at 'unexpected encoding: Utf8Error { valid_up_to: 2, error_len: Some(1) }', .cargo/registry/src/index.crates.io-6f17d22bba15001f/wax-0.5.0/src/lib.rs:90:14
stack backtrace:
   0:     0x5602c3b035c6 - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::hbd7d55b7108d2ab8
   1:     0x5602c21cd7ff - core::fmt::write::h6d54cd7c9e155ec5
   2:     0x5602c3afe471 - std::io::Write::write_fmt::h6a453a71c692f63b
   3:     0x5602c3b033d5 - std::sys_common::backtrace::print::h4ddf81241a51b337
   4:     0x5602c3b04f45 - std::panicking::default_hook::{{closure}}::hff91f1f484ade5cd
   5:     0x5602c3b04d04 - std::panicking::default_hook::h21f14afd59f7aef9
   6:     0x5602c3b05494 - std::panicking::rust_panic_with_hook::h45f66047b14c555c
   7:     0x5602c3b05234 - std::panicking::begin_panic_handler::{{closure}}::h49d1a88ef0908eb4
   8:     0x5602c3b03be6 - std::sys_common::backtrace::__rust_end_short_backtrace::hccebf9e57f8cc425
   9:     0x5602c3b04fb2 - rust_begin_unwind
  10:     0x5602c1d70653 - core::panicking::panic_fmt::h54ec9d0e3180a83d
  11:     0x5602c1d70ac3 - core::result::unwrap_failed::h1cd730365d65235f
  12:     0x5602c3c09bd3 - <core::result::Result<T,E> as wax::ResultExt<T,E>>::expect_encoding::h3ce11cb345e210d4
  13:     0x5602c3c127d6 - core::ops::function::impls::<impl core::ops::function::FnMut<A> for &mut F>::call_mut::hdb0941111fe9e7fb
  14:     0x5602c3c00b9a - <core::iter::adapters::fuse::Fuse<I> as core::iter::traits::iterator::Iterator>::next::ha88050c0f2b5a946
  15:     0x5602c3c00e7e - <itertools::with_position::WithPosition<I> as core::iter::traits::iterator::Iterator>::next::h78759549fc753808
  16:     0x5602c3c13da6 - <wax::walk::Walk as core::iter::traits::iterator::Iterator>::next::h064bb1165bd86fa9
  17:     0x5602c255e465 - nu_command::filesystem::glob::glob_to_value::haed44879b5034d46
  18:     0x5602c255d936 - <nu_command::filesystem::glob::Glob as nu_protocol::engine::command::Command>::run::hb9ee91a73fdb29ed
  19:     0x5602c28a96c6 - nu_engine::eval::eval_call::h6da518bd1c69dbc6
  20:     0x5602c28af412 - nu_engine::eval::eval_expression_with_input::hb9e130153b506b4f
  21:     0x5602c28af9c1 - nu_engine::eval::eval_element_with_input::h32de2f83b1618076
  22:     0x5602c28b0c4f - nu_engine::eval::eval_block::hdd5a1dcf62849732
  23:     0x5602c22c9728 - nu_cli::util::eval_source::hcc5e91169355f6fe
  24:     0x5602c22f443c - nu_cli::repl::evaluate_repl::hd4979d8eefcd4565
  25:     0x5602c229c89c - nu::run::run_repl::h67dc89555da00e30
  26:     0x5602c22adfb1 - nu::main::hc3c3e386e2832857
  27:     0x5602c229ceeb - std::sys_common::backtrace::__rust_begin_short_backtrace::h66bcb1654dd6a60a
  28:     0x5602c229d551 - std::rt::lang_start::{{closure}}::h137ad699032dca5e
  29:     0x5602c3af5d55 - std::rt::lang_start_internal::hf502095b101390bb
  30:     0x5602c22aed35 - main
  31:     0x7f24040456ca - <unknown>
  32:     0x7f2404045785 - __libc_start_main
  33:     0x5602c1de8351 - _start
  34:                0x0 - <unknown>
```

### How to reproduce

create a file with 'broken' enocoding using a different shell, e.g. bash
`touch 'ABCDE'$'\361''DEF'`
launch nu
list files using `glob *`

### Expected behavior

report an error similar like `ls` does

### Configuration

❯ : version | transpose key value | to md --pretty
| key                | value                                                                                                                     |
| ------------------ | ------------------------------------------------------------------------------------------------------------------------- |
| version            | 0.85.1                                                                                                                    |
| branch             | main                                                                                                                      |
| commit_hash        | d51e82f33b17ffc03550822c66025f25cdc1ca40                                                                                  |
| build_os           | linux-x86_64                                                                                                              |
| build_target       | x86_64-unknown-linux-gnu                                                                                                  |
| rust_version       | rustc 1.71.1 (eb26296b5 2023-08-03)                                                                                       |
| rust_channel       | 1.71.1-x86_64-unknown-linux-gnu                                                                                           |
| cargo_version      | cargo 1.71.1 (7f1d04c00 2023-07-29)                                                                                       |
| build_time         | 2023-10-07 10:55:17 +02:00                                                                                                |
| build_rust_channel | release                                                                                                                   |
| allocator          | mimalloc                                                                                                                  |
| features           | dataframe, default, extra, sqlite, trash, which, zip                                                                      |
| installed_plugins  | file, from eml, from ics, from ini, from vcf, gstat, highlight, json path, query, query json, query web, query xml, regex |

**Comment by joergsch** [comment]
changing the dependent version of was from `0.5` to `0.6` in `crates/nu-command/Cargo.toml` seems to solve this issue
I guess it is related to https://github.com/olson-sean-k/wax/issues/26

**Comment by sholderbach** [comment]
Should be closed by #10574
